src/components/popup/transaction-list.tsx

```diff
@@ -1,22 +1,26 @@
-import { createTxDateFormatList } from '@common/group-txs-by-date';
+import { createTxDateFormatList } from '@common/transactions/transaction-utils';
+import { AddressTransactionWithTransfers } from '@stacks/stacks-blockchain-api-types';
 import { Box, Stack, Text, color } from '@stacks/ui';
-import React from 'react';
-import { TxItem } from './tx-item';
+import React, { useMemo } from 'react';
+import { TxWithTransferItem } from './tx-item';
 
-interface TransactionListProps {
-  txsByDate: ReturnType<typeof createTxDateFormatList>;
+interface TransactionWithTransferListProps {
+  txs: AddressTransactionWithTransfers[];
 }
-export function TransactionList({ txsByDate }: TransactionListProps) {
+
+export function TransactionWithTransferList({ txs }: TransactionWithTransferListProps) {
+  // Group txs by date
+  const txsByDate = useMemo(() => (txs ? createTxDateFormatList(txs) : []), [txs]);
```

Comment:
```suggestion
  const txsGroupedByDate = useMemo(() => (txs ? createTxDateFormatList(txs) : []), [txs]);
```

If `txsByDate` isn't clear enough, let's just rename the variable, rather than adding a comment.